contracts/p1/BasketHandler.sol

```diff
@@ -255,7 +255,12 @@ contract BasketHandlerP1 is ComponentP1, IBasketHandler {
 
         for (uint256 i = 0; i < size; ++i) {
             CollateralStatus s = assetRegistry.toColl(basket.erc20s[i]).status();
-            if (s.worseThan(status_)) status_ = s;
+            if (s.worseThan(status_)) {
+                status_ = s;
+            }
+            if (status_ == CollateralStatus.DISABLED) {
```

Comment:
I think this optimization probably increases gas usage on net. There should only ever be a very small number of times this executes with one of the collateral being DISABLED, vs an enormous number of times it executes without DISABLED.